### DISCUSSION TITLE
Require bracketting when detecting floating point size in code notes.

### DISCUSSION THREAD

**Discussion by redwizard42** [RetroAchievements/RAIntegration#971] [Ideas]
Code notes will very greedily take any instances of the letters FLOAT next to each other and treat the address as a 32-bit floating point (single precision) size. Those letters frequently are meant to be a word in the note's description rather than size, which can lead to oddness like "[8bit] Fish Up the Cat floater" being treated as a floating point.  My suggestion would be to only treat those letters as a floating point size if they are the sole letters in a word, and also included in brackets of some kind (any kind).

Summary:
* Exclude any words that are not solely 'float': Floating, Floater, Floated, etc.
* Require some sort of bracketing: parentheses, square brackets to surround it. Ex: (float) [float]

**Comment by Jamiras** [comment]
> Exclude any words that are not solely 'float': Floating, Floater, Floated, etc.

This already occurs. "float" must appear as a singular word, with a non-alphabetic character (or end of string) on both sides. Here are tests cases illustrating:

https://github.com/RetroAchievements/RAIntegration/blob/37ae6dd4dd217a30cc1bc90c609de3c8aafeb177/tests/data/models/CodeNotesModel_Tests.cpp#L232-L238

> "[8bit] Fish Up the Cat floater" being treated as a floating point

That does not happen. "[8bit] Fish Up the Cat float" could be affected. Maybe something could be done to ignore extraneous "float" words if an explicit non-adjacent size exists in the string.

 > Require some sort of bracketing: parentheses, square brackets to surround it. Ex: (float) [float]
 
That would be inconsistent with non-float notations. There are many notes like "16-bit money value" already in the system. Making brackets required would break those.

**Comment by redwizard42** [comment]
> That does not happen. "[8bit] Fish Up the Cat float" could be affected. Maybe something could be done to ignore extraneous "float" words if an explicit non-adjacent size exists in the string.

That's probably worth doing. I think I was thinking of old behavior, thanks to a conversation about something, but preferring other noted sizes sounds worthwhile.
